packages/yew/src/platform/fmt/mod.rs

```diff
@@ -0,0 +1,70 @@
+//! Asynchronous utilities to work with `String`s.
+
+use std::future::Future;
+
+use futures::future::{self, MaybeDone};
+use futures::stream::{FusedStream, Stream};
+use futures::StreamExt;
+use pin_project::pin_project;
+
+mod buffer;
+
+pub(crate) use buffer::{buffer, BufReader, BufWriter};
+
+/// A buffered asynchronous string Stream.
+///
+/// A BufStream combines a BufWriter - BufReader pair and a resolving future that writes to the
+/// buffer and polls the future alongside the buffer.
+#[pin_project]
+pub(crate) struct BufStream<F>
+where
+    F: Future<Output = ()>,
```

Comment:
Perhaps remove the bound from here? That would mean that everywhere this type is named doesn't require the bound - only the usages do

Reply:
We cannot remove this as this bound is declared on the MaybeDone type itself.